athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/querypassthrough/QueryPassthroughSignature.java

```diff
@@ -0,0 +1,162 @@
+        if (!verifyFunctionSignature(engineQptArguments)) {
+            throw new IllegalArgumentException("Function Signature doesn't match implementation's");
+        }
+        //Ensuring the arguments received from the engine are the one defined by the connector
+        for (String argument : this.getFunctionArguments()) {
+            if (!engineQptArguments.containsKey(argument)) {
+                throw new IllegalArgumentException("Missing Query Passthrough Argument: " + argument);
+            }
+            if (StringUtils.isEmpty(engineQptArguments.get(argument))) {
+                throw new IllegalArgumentException("Missing Query Passthrough Value for Argument: " + argument);
+            }
+        }
+        //Finally, perform any connector-specific verification;
+        customConnectorVerifications();
+    }
+
+    /**
+     * Provides a mechanism to perform custom connector verification logic.
+     */
+    default void customConnectorVerifications()
+    {
+        //No Op
+    }
+
+    /**
+     * Verifying that the query passthrough function signature is the one expected by the connector
+     * @param argumentValues
+     * @return true if the signature matches, otherwise false
+     */
+    default boolean verifyFunctionSignature(Map<String, String> argumentValues)
+    {
+        if (argumentValues.containsKey(SCHEMA_FUNCTION_NAME)) {
+            String receivedSignature = argumentValues.get(SCHEMA_FUNCTION_NAME);
+            getLogger().info("Found signature: {}", receivedSignature);
+            return receivedSignature.equalsIgnoreCase(getFunctionSignature());
+        }
+        getLogger().warn("No matching function signature found: {}", getFunctionSignature());
+        return false;
+    }
+
+    /**
+     * Creates a list of Optimization that includes the query passthrough definition
+     * @return list of capability describes of the signature of the current implementation
+     */
+    default List<OptimizationSubType> getQueryPassthroughCapabilities()
+    {
+        List<OptimizationSubType> queryPassthroughDefinition = new ArrayList<>(3);
+        queryPassthroughDefinition.add(QueryPassthrough.QUERY_PASSTHROUGH_SCHEMA.withSchema(getFunctionSchema()));
+        queryPassthroughDefinition.add(QueryPassthrough.QUERY_PASSTHROUGH_NAME.withName(getFunctionName()));
+        queryPassthroughDefinition.add(QueryPassthrough.QUERY_PASSTHROUGH_ARGUMENTS.withArguments(getFunctionArguments()));
+
+        return queryPassthroughDefinition;
+    }
+
+    /**
+     * Adds the query passthrough implementation, if user has not disabled it, to the connector's capabilities
+     * @param capabilities
+     * @param configOptions
+     */
+    default void addQueryPassthroughCapabilityIfEnabled(ImmutableMap.Builder<String, List<OptimizationSubType>> capabilities, Map<String, String> configOptions)
+    {
+        if (allowQueryPassthrough(configOptions)) {
```

Comment:
What is the purpose of disabling this?

Reply:
To allow customer to opt out from enable QPT; customer can add `enable_query_passthrough` env variable to their lambda environment and disable QPT.